File: 1013-fibonacci-number/1013-fibonacci-number.go
func solve(n int, dp []int) int {
    if n == 0 || n == 1 {
		return n
	}
	if dp[n] != 0 {
		return dp[n]
	}
	dp[n] = solve(n - 1, dp) + solve(n - 2, dp)
	return dp[n]
}

	func fib(n int) int {
		dp := make([]int, 31)
		dp[0] = 0
		dp[1] = 1

		return solve(n, dp)
	}

File: 1013-fibonacci-number/1013-fibonacci-number.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {
    const arr = Array(31).fill(-1);
    arr[0] = 0;
    arr[1] = 1;
    const memo = (n) => {
        if (n === 0 || n === 1)
            return n
        if (arr[n] !== -1)
            return arr[n]
        
        return arr[n] = memo(n - 1) + memo(n - 2)
    }
    return memo(n)
};